test/hamiltonian.jl

```diff
@@ -12,6 +12,8 @@ include("common.jl")
     h1 = Hamiltonian(metric, f, g)
     h2 = Hamiltonian(metric, GaussianKinetic(), f, g)
     @test h1 == h2
+    h3 = Hamiltonian(metric, RelativisticKinetic(1.0, 1.0), f, g)
```

Comment:
Let's keep these tests under `experimental/tests/runtests.jl`. We can then add this script to a new CI action.